src/systems/implicit_system.C

```diff
@@ -55,6 +57,8 @@ ImplicitSystem::~ImplicitSystem ()
 {
   // Clear data
   this->clear();
+
+  remove_matrix("System Matrix");
```

Comment:
This, on the other hand, looks like the ideal workaround for what otherwise would have been a memory leak introduced by the improved ImplicitSystem::clear() behavior.